### PR TITLE
PR Builder GitLab Integration

### PR DESCRIPTION
**Pull request by saadmk11**
ref: https://github.com/readthedocs/readthedocs.org/issues/6052

### CODE REVIEW

File: readthedocs/api/v2/views/integrations.py [hunk 1]
```diff
@@ -441,6 +463,17 @@ def is_payload_valid(self):
             return False
         return token == secret
 
+    def get_external_version_data(self):
+        """Get Commit Sha and merge request number from payload."""
```
Comment (ericholscher) [inline readthedocs/api/v2/views/integrations.py]:
```suggestion
        """Get commit SHA and merge request number from payload."""
```

File: readthedocs/api/v2/views/integrations.py [hunk 2]
```diff
@@ -441,6 +463,17 @@ def is_payload_valid(self):
...
+        try:
+            identifier = self.data['object_attributes']['last_commit']['id']
+            verbose_name = str(self.data['object_attributes']['iid'])
+
+            return identifier, verbose_name
+
+        except KeyError:
+            raise ParseError('Parameters "id" and "iid" are required')
```
Comment (ericholscher) [inline readthedocs/api/v2/views/integrations.py]: Where is this `ParseError` handled?

File: readthedocs/rtd_tests/tests/test_project.py [hunk 3]
```diff
@@ -193,6 +195,21 @@ def test_get_latest_build_excludes_external_versions(self):
         # Test that External Version is not considered for get_latest_build.
         self.assertEqual(self.pip.get_latest_build(), None)
 
+    def test_git_provider_name_github(self):
+        self.assertEqual(self.pip.git_provider_name, GITHUB_BRAND)
```
Comment (ericholscher) [inline readthedocs/rtd_tests/tests/test_project.py]: These should probably explicitly assign the `repo`, like the `gitlab` tests, so they won't fail if we change something elsewhere.

File: readthedocs/api/v2/views/integrations.py (hunk 2 above)
Comment (saadmk11) [reply readthedocs/api/v2/views/integrations.py]: When there is a `ParseError` we return `400` to the API. We do the same thing for the `push` events.

File: readthedocs/rtd_tests/tests/test_project.py (hunk 3 above)
Comment (saadmk11) [reply readthedocs/rtd_tests/tests/test_project.py]: Done!

**Pull request merged**